webpack.config.js

```diff
@@ -0,0 +1,179 @@
+    },
+    devtool: 'inline-source-map',
+    module: {
+        rules: [
+            {
+                test: /\.[jt]sx?$/,
+                exclude: /node_modules/,
+                use: [
+                    {
+                        loader: "babel-loader",
+                        options: createBabelConfig({ mode: argv.mode }),
+                    },
+                ],
+            },
+            {
+                test: /\.css$/,
+                exclude: /node_modules/,
+                use: ["style-loader", "css-loader"],
+                sideEffects: true,
+            },
+            {
+                test: /\.(scss)$/,
+                use: ["style-loader", "css-loader", "postcss-loader", "fast-sass-loader"],
+                sideEffects: true,
+            },
+            {
+                test: /\.(png|jpe?g|gif|svg)$/i,
+                loader: 'file-loader',
+            },
+        ],
+    },
+
+    resolve: {
+        extensions: ['.tsx', '.ts', '.jsx', '.js'],
+        // polyfill
+        fallback: {
+            http: require.resolve("stream-http"),
+            https: require.resolve("https-browserify"),
+            buffer: require.resolve("buffer"),
+            util: require.resolve("util"),
+            process: require.resolve("process/browser"),
+        },
+    },
+
+    output: {
+        path: path.resolve(__dirname, "./dist"), // cwp cant read this, it's dumb (even though this is the default as of wbpack5)
+        filename: `[name].bundle.js`,
+        chunkLoadTimeout: 15000,
+    },
+    cache: {
+        type: 'filesystem',
+        cacheLocation: path.resolve(__dirname, `.yarn/webpack/${argv.mode}`),
+        buildDependencies: {
+            config: [__filename],
+        },
+    },
+
+    optimization: {
+        emitOnErrors: false,
+        minimizer: [
+            new TerserPlugin({
+                extractComments: false,
+                terserOptions: {
+                    output: {
+                        ascii_only: true,
+                        comments: false,
+                    },
+                },
+            }),
+        ],
+        splitChunks: {
+            chunks: "all",
+            cacheGroups: {
+                packages: {
+                    test: /[\\/]node_modules[\\/]/,
+                    idHint: 'packages',
+                },
+                api: {
+                    priority: 1,
+                    test: /[\\/]ApiClient[\\/]/,
+                    idHint: 'api',
+                    enforce: true,
+                },
+                styles: {
+                    test: /\.(scss)$/,
+                    idHint: 'styles',
+                    enforce: true,
+                },
+            },
+        },
+    },
+
+    devServer: {
+        static: path.join(__dirname, "dist"),
+        compress: true,
+        hot: true,
+        host: "0.0.0.0", // technically insecure? don't put nudes in your app, helps to test with mobile
+        port: 5000,
```

Comment:
Considering 5000 is TGS default port, please dont

```suggestion
        port: 8080,
```